### PR TITLE
BnA (inference) fusion + some minor updates for CBA fusion

### PR DESCRIPTION
**Pull request by deven-amd**
This PR adds support for the     BatchNorm+Activation (inference) fusion

The implementation leverages the fused BnA inference kernel provided by the MIOpen fusion API. Currently only the float datatype is supported. Support for BnA (training) and for fp16 datatype, will be added once the same becomes available in the MIOpen fusion API.

Set the env var TF_ROCM_FUSION_ENABLE=1 to enable the fusion.

Currently fusion related VLOG messages will get printed by default, when fusion is enabled. This will be disabled in the near future.

---------------------------------

The following env vars are temporary, used primarily for debug purposes. The functionality associated with them is not expected to persist long term.

1. Set the env var TF_ROCM_FUSION_PASS_POST_PARTITION=1 to change the grouping where the ROCM Fusion Pass runs. By default it runs as part of the "post placement" grouping (i.e. after op placement has been done, but before the graph has been partitioned into per device subgraphs). Setting the env var will change the grouping to "post partitioning". 

2. Set the env var TF_ROCM_FUSION_DUMP_GRAPH_BEFORE=1 to dump the TF graph to stdout *before* the ROCm FUSION Pass is run

3. Set the env var TF_ROCM_FUSION_DUMP_GRAPH_AFTER=1 to dump the TF graph to stdout *after* the ROCm FUSION Pass is run

### CODE REVIEW

File: tensorflow/core/graph/gpu_fusion_pass.cc [hunk 1]
```diff
@@ -48,19 +48,24 @@ namespace gpu_fusion_pass {
 
 // declaration of all the constants that are used in the gpu_fusion_pass
 // namespace
-const int kVlogLevel = 2;
+const int kVlogLevel = -1;
 
 // graph pass grouping for this fusion pass
 const OptimizationPassRegistry::Grouping kROCmFusionPassGrouping =
-    OptimizationPassRegistry::POST_PLACEMENT;
+    (getenv("TF_ROCM_FUSION_PASS_POST_PARTITIONING") == nullptr)
```
Comment (whchung) [inline tensorflow/core/graph/gpu_fusion_pass.cc]: @deven-amd could you update the description to say more about this env var?

File: tensorflow/core/graph/gpu_fusion_pass.cc [hunk 2]
```diff
@@ -316,7 +341,9 @@ Status ROCmFusionPass::Run(const GraphOptimizationPassOptions& options) {
 }
 
 bool ROCmFusionPass::RunPass(Graph* graph) {
-  // DumpGraph("Before running ROCmFusionPass", &**g);
+  if (getenv("TF_ROCM_FUSION_DUMP_GRAPH_BEFORE")) {
```
Comment (whchung) [inline tensorflow/core/graph/gpu_fusion_pass.cc]: @deven-amd also please help enumerate env vars introduced in this PR in the description so other users can leverage them effectively.

File: tensorflow/core/graph/gpu_fusion_pass.cc [hunk 3]
```diff
@@ -394,6 +423,7 @@ bool ROCmFusionOpConvolutionBiasBatchNormActivation::IsFusionEligible(
           d->bias = n2;
           d->norm = n3;
           d->actv = n4;
+          VLOG(kVlogLevel) << "===========";
```
Comment (whchung) [inline tensorflow/core/graph/gpu_fusion_pass.cc]: remove this?

File: tensorflow/core/graph/gpu_fusion_pass.cc [hunk 4]
```diff
@@ -447,6 +477,7 @@ bool ROCmFusionOpConvolutionBiasActivation::IsFusionEligible(const Node* n3,
         d->conv = n1;
         d->bias = n2;
         d->actv = n3;
+        VLOG(kVlogLevel) << "===========";
```
Comment (whchung) [inline tensorflow/core/graph/gpu_fusion_pass.cc]: remove this line?

File: tensorflow/core/graph/gpu_fusion_pass.cc (hunk 3 above)
Comment (deven-amd) [reply tensorflow/core/graph/gpu_fusion_pass.cc]: thats by design to make looking at the VLOG output easier on the eye...would like to leave it in

File: tensorflow/core/graph/gpu_fusion_pass.cc (hunk 4 above)
Comment (deven-amd) [reply tensorflow/core/graph/gpu_fusion_pass.cc]: same as above

File: tensorflow/core/graph/gpu_fusion_pass.cc (hunk 3 above)
Comment (whchung) [reply tensorflow/core/graph/gpu_fusion_pass.cc]: yeah I looked at your other codes down below and it appears to be that way. ok to leave it as-is.
